cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/NamespaceHttpHandler.java

```diff
@@ -162,7 +162,8 @@ public void create(HttpRequest request, HttpResponder responder, @PathParam("nam
   @Path("/unrecoverable/namespaces/{namespace-id}")
   public void delete(HttpRequest request, HttpResponder responder, @PathParam("namespace-id") String namespace) {
     if (!cConf.getBoolean(Constants.Dangerous.UNRECOVERABLE_RESET, Constants.Dangerous.DEFAULT_UNRECOVERABLE_RESET)) {
-      responder.sendStatus(HttpResponseStatus.FORBIDDEN);
+      responder.sendString(HttpResponseStatus.FORBIDDEN,
+                           "Namespace deletion is not enabled. Please enable it and restart CDAP master.");
```

Comment:
I'm not sure about that. I thought about it, but to me, it is a dangerous property to expose as part of an HTTP response. If users really wish to delete, they should look at documentation?